pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -281,44 +283,32 @@ protected void setupInitialSegments(String realtimeTableName, ZNRecord partition
     List<String> paths = new ArrayList<>(nPartitions);
     List<ZNRecord> records = new ArrayList<>(nPartitions);
     final long now = System.currentTimeMillis();
-    final int seqNum = 0; // Initial seq number for the segments
+    final int seqNum = STARTING_SEQUENCE_NULMBER;
 
     for (int i = 0; i < nPartitions; i++) {
-      SimpleConsumerWrapper kafkaConsumer = SimpleConsumerWrapper.forPartitionConsumption(new KafkaSimpleConsumerFactoryImpl(),
-          bootstrapHosts, "dummyClientId", topicName, i);
-      try {
-        final List instances = partitionMap.get(Integer.toString(i));
-        LLCRealtimeSegmentZKMetadata metadata = new LLCRealtimeSegmentZKMetadata();
-        final String rawTableName = TableNameBuilder.extractRawTableName(realtimeTableName);
-        LLCSegmentName llcSegmentName = new LLCSegmentName(rawTableName, i, seqNum, now);
-        final String segName = llcSegmentName.getSegmentName();
-
-        metadata.setCreationTime(now);
-
-        final long startOffset;
-        try {
-          startOffset = kafkaConsumer.fetchPartitionOffset(initialOffset, KAFKA_PARTITION_OFFSET_FETCH_TIMEOUT_MILLIS);
-        } catch (TimeoutException e) {
-          LOGGER.warn("Timed out when fetching partition offsets for segment {}", segName);
-          throw new RuntimeException(e);
-        }
-        LOGGER.warn("Setting start offset for segment {} to {}", segName, startOffset);
-        metadata.setStartOffset(startOffset);
-        metadata.setEndOffset(Long.MAX_VALUE);
-
-        metadata.setNumReplicas(instances.size());
-        metadata.setTableName(rawTableName);
-        metadata.setSegmentName(segName);
-        metadata.setStatus(CommonConstants.Segment.Realtime.Status.IN_PROGRESS);
-
-        ZNRecord record = metadata.toZNRecord();
-        final String znodePath = ZKMetadataProvider.constructPropertyStorePathForSegment(realtimeTableName, segName);
-        paths.add(znodePath);
-        records.add(record);
-        idealStateEntries.put(segName, instances);
-      } finally {
-        IOUtils.closeQuietly(kafkaConsumer);
-      }
+      final List instances = partitionMap.get(Integer.toString(i));
```

Comment:
partitionMap renamed to  partitionToServersMap

List changed to List<String>

No, the instances are always populated when the partition assignment is done. If no instances are present, table creation will throw exception (in the partition assignment code). When the table is removed, the entire partition assignment znode is removed as well.